ethers-core/src/utils/mod.rs

```diff
@@ -421,6 +421,27 @@ pub fn to_checksum(addr: &Address, chain_id: Option<u8>) -> String {
     })
 }
 
+/// checks if the given address is a valid checksum address
+pub fn verify_checksum(addr: &str, chain_id: Option<u8>) -> bool {
+    let addr = addr.strip_prefix("0x").unwrap_or(addr);
+    let address = addr.parse::<Address>();
+
+    // wrong address string
+    if address.is_err() {
+        return false
+    }
+
+    // addr is checksummed address
+    if addr.chars().any(|c| c.is_uppercase()) {
+        let checksum_addr = to_checksum(&address.unwrap(), chain_id);
+
+        return checksum_addr.strip_prefix("0x").unwrap_or(&checksum_addr) == addr
+    }
+
+    // addr is not checksummed address
+    true
```

Comment:
this should be false, no?

Reply:
maybe the function name cause this confuse.
if the address is checksummed address, it need to do validation.
but if the address is not checksummed, it should not to do validation, so I chose to return `true`